Prevent crash when only D0 channel is set, D1-D3 set to None

dueringa opened the pull request:
When channels are set undefined / None, is_used should be false in AddChannel call.

ewfuentes commented:
Thank you for the contribution! I've made you a collaborator, so feel free to merge whenever you think that your changes are ready.

dueringa commented:
Thanks. This is actually part of a bigger bug, which leads Logic.exe to crash (under some circumstances?) when DAT1-DAT3 are are set to NONE. I assume the problem is in SDIOAnalyzer.cpp, lines 62-64:

```	
        mDAT1 = GetAnalyzerChannelData(mSettings->mDAT1Channel);
	mDAT2 = GetAnalyzerChannelData(mSettings->mDAT2Channel);
	mDAT3 = GetAnalyzerChannelData(mSettings->mDAT3Channel);
```

However, if you don't set these, all other references mustn't use mDAT1-mDAT3 either.

dueringa commented:
I think I'm finished with this. 

I added some checks in the Settings to prevent invalid data channel selections. Also, I fixed a crash of the Logic software when only DAT0 was set, DAT1-DAT3 were set to None.

Can you have a look if you're okay with this? (I don't really like if-statements without curly braces, but the code might get more unclear with separate 4-line if blocks for each data line for each `AdvanceToAbsPosition`)
